pyvista/core/filters/image_data.py

```diff
@@ -1604,3 +1608,293 @@ def _update_and_get_output():
+        >>> _ = pl.add_mesh(connected.threshold(0.5), show_edges=True)
+        >>> _ = pl.add_mesh(
+        ...     connected.threshold(0.5, invert=True),
+        ...     show_edges=True,
+        ...     opacity=0.5,
+        ... )
+        >>> pl.show()
+
+        Label all connected regions with a constant value.
+
+        >>> connected, labels, sizes = segmented_grid.label_connectivity(
+        ...     label_mode='constant', constant_value=10
+        ... )
+        >>> pl = pv.Plotter()
+        >>> _ = pl.add_mesh(connected.threshold(0.5), show_edges=True)
+        >>> _ = pl.add_mesh(
+        ...     connected.threshold(0.5, invert=True),
+        ...     show_edges=True,
+        ...     opacity=0.5,
+        ... )
+        >>> pl.show()
+
+        Label only the regions that include seed points, by seed order.
+
+        >>> points = [(2, 1, 0), (0, 0, 1)]
+        >>> connected, labels, sizes = segmented_grid.label_connectivity(
+        ...     extraction_mode='seeded', point_seeds=points
+        ... )
+        >>> pl = pv.Plotter()
+        >>> _ = pl.add_mesh(connected.threshold(0.5), show_edges=True)
+        >>> _ = pl.add_mesh(
+        ...     connected.threshold(0.5, invert=True),
+        ...     show_edges=True,
+        ...     opacity=0.5,
+        ... )
+        >>> pl.show()
+
+        """
+        # Get a copy of input to not overwrite data
+        input_mesh = self.copy()  # type: ignore[attr-defined]
+
+        if scalars is None:
+            set_default_active_scalars(input_mesh)
+        else:
+            input_mesh.set_active_scalars(scalars)
+
+        # Make sure we have point data (required by the filter)
+        field, scalars = input_mesh.active_scalars_info
+        if field == FieldAssociation.CELL:
+            # Convert to point data
+            input_mesh = input_mesh.cells_to_points(scalars=scalars, copy=False)
+
+        scalars_casted_to_float = False
+        if force_float and np.issubdtype(input_mesh.point_data[scalars].dtype, np.integer):
+            input_mesh.point_data[scalars] = input_mesh.point_data[scalars].astype(float)
+            # Keep track of the operation to cast back to int when the operation is inplace
+            scalars_casted_to_float = True
+
+        # Set vtk algorithm
+        alg = _vtk.vtkImageConnectivityFilter()
+        alg.SetInputDataObject(input_mesh)
+        alg.SetInputArrayToProcess(
+            0,
+            0,
+            0,
+            field.value,
+            scalars,
+        )  # args: (idx, port, connection, field, name)
+
+        # Set the scalar range considered for connectivity
+        # Default to vtk default 0.5 to VTK_DOUBLE_MAX, nothing to do
+        # See https://vtk.org/doc/nightly/html/classvtkImageConnectivityFilter.html
+        if scalar_range is not None:
+            if isinstance(scalar_range, str) and scalar_range == 'auto':
+                unique_scalars = np.unique(input_mesh.point_data[scalars])
```

Comment:
I like your default of using the full data range. This matches how the `connectivity` filter also works by default. I think setting to `None` for the vtk default a bit odd in term of API, since actively setting to `None` is actually setting a range, not remove it.

What about using `None` to mean `'auto'` instead? I.e. use `None` to use the full data range (by default). And then use the term `foreground` to use the VTK default? extracting foreground connectivity is a common use case in image processing and is effectively what the VTK default is doing with `[0.5, VTK_DOUBLE_MAX]`. 

Feel free to disagree though and simply mark this as resolved and keep using `'auto'`.

Reply:
Agreed, explicit is better.

So now there are 3 different keywords:
- `auto`: uses the full range, with the same behaviour as `connectivity` filter
- `foreground`: automatically determine the foreground using the full range minus the smallest value. This is very similar to the default vtk value but more flexible, allowing to not cast the data to floats.
- `vtk_default`